cpp/src/arrow/compute/kernels/cast.h

```diff
@@ -56,6 +58,7 @@ struct ARROW_EXPORT CastOptions {
   bool allow_time_truncate;
   bool allow_time_overflow;
   bool allow_float_truncate;
+  bool allow_decimal_truncate;
```

Comment:
You shouldn't add this if this isn't used anywhere.

Reply:
removed.

Reply:
Actually it is used now